internal/core/item_tree.rs

```diff
@@ -785,6 +785,31 @@ impl ItemRc {
     ) -> Option<R> {
         self.visit_descendants_impl(&mut visitor)
     }
+
+    /// Returns the transform to apply to children to map them into the local coordinate space of this item.
+    /// Typically this is the identity, but rotation for example may change this.
+    #[cfg(not(slint_int_coord))]
+    pub fn children_transform(&self) -> ItemTransform {
+        self.downcast::<crate::items::Rotate>().map_or_else(
+            ItemTransform::identity,
+            |rotate_item| {
+                let origin = crate::lengths::LogicalVector::from_lengths(
+                    rotate_item.as_pin_ref().rotation_origin_x(),
+                    rotate_item.as_pin_ref().rotation_origin_y(),
+                );
+                ItemTransform::translation(-origin.x, -origin.y)
+                    .then_rotate(euclid::Angle {
+                        radians: rotate_item.as_pin_ref().rotation_angle().to_radians(),
+                    })
+                    .then_translate(origin)
+            },
+        )
+    }
+
+    #[cfg(slint_int_coord)]
```

Comment:
An alternative to this would be to explicitly use f32 and try to round when computing the dirty rects.

Reply:
> An alternative to this would be to explicitly use f32 and try to round when computing the dirty rects.

Because a transform in integer doesn't make sense at all anyway.
I would suggest to hardcode to f32 and do `cast()` when operating in the transform.

Reply:
Cool. That's what I thought. Makes the int case a little slower though.

Reply:
Or perhaps a fixed point type, but that can also be messy